src/series25.rs

```diff
@@ -222,3 +266,24 @@ impl<SPI: Transfer<u8>, CS: OutputPin> BlockDevice<u32, SPI, CS> for Flash<SPI,
         Ok(())
     }
 }
+
+// TODO reintroduce test
+// Manually ran this test by tweaking [dev-dependencies] and config;
+// however, until we can specify separate dependencies for
+// tests and examples, this will not be runnable in automation.
```

Comment:
This looks good, what's the issue with the test? Normally `no_std` libraries can be made testable by replacing the `#![no_std]` with `#![cfg_attr(not(test), no_std)]`.

Reply:
The issue is apparently that `[dev-dependencies]` applies to both test and the `dump` example, which pulls in eg `stm32f4xx-hal`, and there's no way to define separate dependencies for test.  `series25.rs` doesn't include `#![no_std]` currently.  If you simply include this one test and run:
`cargo test --package spi-memory --lib series25::tests`

You get the error:
```
error[E0463]: can't find crate for `test`
   --> src/series25.rs:280:5
    |
280 | /     fn test_decode_jedec_id() {
281 | |         let cypress_id_bytes = [0x7F, 0x7F, 0x7F, 0x7F, 0x7F, 0x7F, 0xC2, 0x22, 0x08];
282 | |         let ident = Identification::from_jedec_id(&cypress_id_bytes);
283 | |         assert_eq!(0xC2, ident.mfr_code());
...   |
287 | |         assert_eq!(device_id[1], 0x08);
288 | |     }
    | |_____^ can't find crate
```